src/main/scala/workflow/graph/ExtractSaveablePrefixes.scala

```diff
@@ -0,0 +1,20 @@
+package workflow.graph
+
+/**
+ * Extract the prefixes of all Nodes whose state we want to save for reuse by other Pipeline apply and fit calls.
+ * This is all nodes that either have a Cacher or an EstimatorOperator as the internal operator.
+ */
+object ExtractSaveablePrefixes extends Rule {
```

Comment:
Just to make sure, for the `Cacher` even if Spark evicts the RDDs this just means that we will still operate with the RDDs on which cache was called. Is that right ?